src/main/java/net/sf/marineapi/nmea/event/AbstractSentenceListener.java

```diff
@@ -101,9 +101,15 @@ public void readingStopped() {
 	@SuppressWarnings("unchecked")
 	public void sentenceRead(SentenceEvent event) {
 		Sentence sentence = event.getSentence();
-		Class<?>[] interfaces = sentence.getClass().getInterfaces();
-		if (Arrays.asList(interfaces).contains(this.expectedType)) {
-			sentenceRead((T) sentence);
+		if (this.expectedType instanceof Class<?>) {
```

Comment:
Right, the hard-coded 0 index is a known risk I took simply because at the time I didn't come up with easy solution that would handle all combinations. Also, I thought such cleverness would be minor use case and most would be using AbstractSentenceListener directly. 

So yes, this is something that could be improved, but let's open a separate issue for it.  Is it a big problem for your app design at the moment or would it be for general improvement?

Until fixed, I guess the problem can still be worked around simply by switching the order of generic types, i.e. `MyClass<A extends Sentence, B>`.

Anyways, returning back to the original question on the if-else in event handler method; as it does what it does, regardless how Type is resolved from generic types, I'd like to get rid of the `instanceof` and probably unnecessary else-branch, assuming the `isAssignableFrom` is always sufficient method for comparison. Do you agree?